File: src/main/java/com/battletrident/games/player/PlayerManager.java
package com.battletrident.games.player;

import org.bukkit.entity.Player;
import org.bukkit.potion.PotionEffect;
import org.bukkit.potion.PotionEffectType;

import java.util.List;

public class PlayerManager {
	private Player[] players;

	public void giveImmune(Player player, boolean isPermanent) {
		player.addPotionEffects(
			List.of(
				new PotionEffect(
					PotionEffectType.RESISTANCE,
					isPermanent?
						PotionEffect.INFINITE_DURATION
						:
						20 * 10,
					Integer.MAX_VALUE,
					true, false
				),
				new PotionEffect(
					PotionEffectType.REGENERATION,
					isPermanent?
						PotionEffect.INFINITE_DURATION
						:
						20 * 10,
					Integer.MAX_VALUE,
					true, false
				)
			)
		);
	}
	public void giveImmune(Player player) {
		giveImmune(player, false);
	}
}

File: src/main/kotlin/com/BattleTrident/games/player/PlayerManager.kt
package com.BattleTrident.games.player

import com.BattleTrident.Utils.onlinePlayers
import org.bukkit.entity.Player

class PlayerManager {
	private val players = HashMap<Player, ArrayList<String>>()

	init {
		for (player in onlinePlayers) {
			players[player] = arrayListOf()
		}
	}

	fun setSkills(player: Player, skills: List<String>) {
		players[player]?.addAll(skills)
	}

	fun getSkills(player: Player): List<String>? {
		return players[player]
	}
}